### PR TITLE
fix(quasar-cli-vue3): fixed prod builds

### PR DESCRIPTION
**Pull request by lexasq**
Fix for https://github.com/module-federation/module-federation-examples/issues/3173

### CODE REVIEW

**Review by CaioRolla — APPROVED** [review]

File: quasar-cli-vue3-webpack-javascript/app-exposes/.quasar/client-entry.js [hunk 1]
```diff
@@ -49,7 +49,10 @@ console.info('[Quasar] Running SPA.')
 
 
 
-const publicPath = ``
+const publicPath = `http://localhost:3001/`
+
+const doubleSlashRE = /\/\//
+const addPublicPath = url => (publicPath + url).replace(doubleSlashRE, '/')
```
Comment (brunos3d) [inline quasar-cli-vue3-webpack-javascript/app-exposes/.quasar/client-entry.js:55]: Just asking shall we use global replace/replace all occurrences here?

Comment (lexasq) [reply quasar-cli-vue3-webpack-javascript/app-exposes/.quasar/client-entry.js:55]: I believe it's autogenerated folder

**Comment by goldyfruit** [comment]
Just tried the PR, when trying to `quasar dev` on `app-general` I got this:

image

**Comment by viktoriialurie** [comment]
@goldyfruit the readme has been changed, could you please try to check with the updated instructions?

**Comment by goldyfruit** [comment]
> @goldyfruit the readme has been changed, could you please try to check with the updated instructions?

This is the error I got with the new README

image

**Comment by lexasq** [comment]
> > @goldyfruit the readme has been changed, could you please try to check with the updated instructions?
> 
> This is the error I got with the new README
> 
> image

Hi, could you please share content of your env file?

**Comment by lexasq** [comment]
@goldyfruit I think I understood your struggle, the updates were made per app READMEs, duplicated those changes to common file as well, hopefully that'll help.

**Comment by goldyfruit** [comment]
> @goldyfruit I think I understood your struggle, the updates were made per app READMEs, duplicated those changes to common file as well, hopefully that'll help.

Indeed, following the new README allows me to run `quasar dev`.

image

The build instructions failed on `app-general`, complaining about `Loading failed for the <script> with source “http://localhost:4000/js/main.29127d47.js”.`

image

Thanks for your help @lexasq

**Comment by goldyfruit** [comment]
Issues seems related to some files with port `4000` referenced in `app-general`.

- `.quasar/client-entry.js`
- `envs/env.prod.json`
- `envs/env.json`

**Comment by lexasq** [comment]
@goldyfruit looks weird, just retested that and its fine. As for the build you got to check the respective instruction in the apps. 



image

image

**Comment by goldyfruit** [comment]
> Issues seems related to some files with port `4000` referenced in `app-general`.
> 
> * `.quasar/client-entry.js`
> * `envs/env.prod.json`
> * `envs/env.json`

@lexasq After changing port `4000` to `3000` in the files listed above and re-run `quasar build` its working.

**Comment by lexasq** [comment]
> > Issues seems related to some files with port `4000` referenced in `app-general`.
> > 
> > * `.quasar/client-entry.js`
> > * `envs/env.prod.json`
> > * `envs/env.json`
> 
> @lexasq After changing port `4000` to `3000` in the files listed above and re-run `quasar build` its working.

Don't change anything manually, follw the READMEs

**Comment by goldyfruit** [comment]
> Don't change anything manually, follw the READMEs

Did it again to make sure _(with a fresh clone from your branch)_, I followed the instructions from the README for both `app-expose` and `app-general`:

```
yarn
yarn lint
yarn format
yarn setup-env:prod
quasar build
```

Same result:

image

**Comment by lexasq** [comment]
> > Don't change anything manually, follw the READMEs
> 
> Did it again to make sure _(with a fresh clone from your branch)_, I followed the instructions from the README for both `app-expose` and `app-general`:
> 
> ```
> yarn
> yarn lint
> yarn format
> yarn setup-env:prod
> quasar build
> ```
> 
> Same result:
> 
> image

You most definitely didin't, `app-general` is set to be hosted on port 4000.

**Comment by goldyfruit** [comment]
> @goldyfruit looks weird, just retested that and its fine. As for the build you got to check the respective instruction in the apps.
> 
> image
> 
> image

But in this screenshot you are running on port `3000`, I'm confused.

**Comment by goldyfruit** [comment]
> > @goldyfruit looks weird, just retested that and its fine. As for the build you got to check the respective instruction in the apps.
> > image
> > image
> 
> But in this screenshot you are running on port `3000`, I'm confused.

When running `quasar serve` on port `4000` its working. :+1:

**Comment by goldyfruit** [comment]
One side note, maybe the main README should be updated with the correct ports as well:

image

Thanks again for your help *(and patience)* @lexasq

**Comment by lexasq** [comment]
[Uploading quasar.webm…]()
@goldyfruit

**Comment by lexasq** [comment]
> One side note, maybe the main README should be updated with the correct ports as well:
> 
> image
> 
> Thanks again for your help _(and patience)_ @lexasq

You can change port in the env file for prod, just don't forget to use the correct command for quasar, as for the dev you change port in the config file.

**Comment by goldyfruit** [comment]
> > One side note, maybe the main README should be updated with the correct ports as well:
> > image
> > Thanks again for your help _(and patience)_ @lexasq
> 
> You can change port in the env file for prod, just don't forget to use the correct command for quasar, as for the dev you change port in the config file.

Thanks!

File: quasar-cli-vue3-webpack-javascript/app-exposes/README.md [hunk 2]
```diff
@@ -37,6 +37,8 @@ npm run format
 ```bash
 yarn setup-env:prod
 quasar build
+cd dist spa
```
Comment (goldyfruit) [inline quasar-cli-vue3-webpack-javascript/app-exposes/README.md]: Should be `dist/spa`.

File: quasar-cli-vue3-webpack-javascript/app-general/README.md [hunk 3]
```diff
@@ -37,6 +37,8 @@ npm run format
 ```bash
 yarn setup-env:prod
 quasar build
+cd dist spa
```
Comment (goldyfruit) [inline quasar-cli-vue3-webpack-javascript/app-general/README.md]: Should be `dist/spa` as well.

Comment (lexasq) [reply quasar-cli-vue3-webpack-javascript/app-general/README.md]: Thanks, overlooked that!

**Review by goldyfruit — APPROVED** [review]

**Pull request merged**